src/platform_impl/linux/wayland/state.rs

```diff
@@ -148,13 +152,17 @@ impl WinitState {
                 (None, None)
             };
 
+        let shm = Shm::bind(globals, queue_handle).map_err(WaylandError::Bind)?;
```

Comment:
Besides, I wonder if can just use `Lazy` here and move the `Arc<GlobalList>` and a `QueueHandle` inside it, so once you call it it'll `bind` the global instead of carrying it around. That way it should be more ergonomic? Also, I think you can `move` the `GlobalList` into it, but it could be tricky.

In general, I'd suggest to try with `Lazy` since it'll simply code and it'll be more clear.

Reply:
I don't get it. How do I bind it exacly?

I can't really create an `Shm` like this. Doesn't it need `impl ShmHandler`? And we anyway have an `Shm` already.
```
let globals = Arc::new(globals);
let globals_clone = globals.clone();
let queue_handle_clone = queue_handle.clone();
let pool = Lazy::new(Box::new(move || {
    let shm = Shm::bind(&globals_clone, &queue_handle_clone).unwrap();
    Arc::new(Mutex::new(SlotPool::new(2, &shm).unwrap()))
}) as _);
```

Reply:
Hm, yeah, we need `Shm` already, sorry. My main issue is that I don't want to create a pool until it really needs to be created. I guess it doesn't work like that with global rebind.

Maybe keep it as is(without any Lazy stuff) and I'll look into it myself.